cmip6_downscaling/workflows/utils.py

```diff
@@ -598,9 +601,10 @@ def rechunk_zarr_array_with_caching(
                 temp_store=temp_store,
             )
             rechunk_plan.execute(retries=5)
-        rechunked_ds = xr.open_zarr(
-            target_store
-        )  # ideally we want consolidated=True but it seems that functionality isn't offered in rechunker right now
+        rechunked_ds = xr.open_zarr(target_store)
+        # remove any temp stores created by task
+        remove_stores([temp_store])
```

Comment:
i assume you've tested that this works alright? like the `temp_store` doesn't end up getting used by rechunker lazily later on, right?